pkg/config/node.go

```diff
@@ -417,34 +428,42 @@ func getNodeIpForRequestedIpStack(node v1.Node, filterIps []string, machineNetwo
 // apiPort: The port on which the k8s api listens. Should be 6443.
 // lbPort: The port on which haproxy listens.
 // statPort: The port on which the haproxy stats endpoint listens.
-func GetConfig(kubeconfigPath, clusterConfigPath, resolvConfPath string, apiVips, ingressVips []net.IP, apiPort, lbPort, statPort uint16) (node Node, err error) {
-	vipCount := 0
-	if len(apiVips) > len(ingressVips) {
-		vipCount = len(apiVips)
-	} else {
-		vipCount = len(ingressVips)
-	}
-	nodes := []Node{}
-	var apiVip, ingressVip net.IP
-	for i := 0; i < vipCount; i++ {
-		if i < len(apiVips) {
-			apiVip = apiVips[i]
-		} else {
-			apiVip = nil
-		}
-		if i < len(ingressVips) {
-			ingressVip = ingressVips[i]
+// clusterLBConfig: A struct containing IPs for API, API-Int and Ingress LBs
+func GetConfig(kubeconfigPath, clusterConfigPath, resolvConfPath string, apiVips, ingressVips []net.IP, apiPort, lbPort, statPort uint16, clusterLBConfig ClusterLBConfig) (node Node, err error) {
+	switch {
```

Comment:
This switch-case adds a level of indentation to the whole function where it's not really needed (i.e. right now the whole body of this function minus single line is written inside an `else`/`default` statement).

As you are immediately returning after your check you can simplify it so instead of

```
switch {
  case XXX:
    return blablabla
  default:
    lots of code
```

you can write

```
if XXX:
  return blablabla
lots of code
```

(note I am skipping `else` completely in my revised version).